core/src/main/scala/org/apache/gearpump/cluster/master/AppManager.scala

```diff
@@ -231,35 +231,38 @@ private[cluster] class AppManager(kvService: ActorRef, launcher: AppMasterLaunch
       timeStamp: TimeStamp, error: Throwable): Unit = {
     applicationRegistry.get(appId) match {
       case Some(appRuntimeInfo) =>
-        var updatedStatus: ApplicationRuntimeInfo = null
-        LOG.info(s"Application $appId change to ${newStatus.toString} at $timeStamp")
-        newStatus match {
-          case ApplicationStatus.ACTIVE =>
-            updatedStatus = appRuntimeInfo.onAppMasterActivated(timeStamp)
-            sender ! AppMasterActivated(appId)
-          case succeeded@ApplicationStatus.SUCCEEDED =>
-            killAppMaster(appId, appRuntimeInfo.worker)
-            updatedStatus = appRuntimeInfo.onFinalStatus(timeStamp, succeeded)
-            appResultListeners.getOrElse(appId, List.empty).foreach{ client =>
-              client ! ApplicationSucceeded(appId)
-            }
-          case failed@ApplicationStatus.FAILED =>
-            killAppMaster(appId, appRuntimeInfo.worker)
-            updatedStatus = appRuntimeInfo.onFinalStatus(timeStamp, failed)
-            appResultListeners.getOrElse(appId, List.empty).foreach{ client =>
-              client ! ApplicationFailed(appId, error)
-            }
-          case terminated@ApplicationStatus.TERMINATED =>
-            updatedStatus = appRuntimeInfo.onFinalStatus(timeStamp, terminated)
-          case status =>
-            LOG.error(s"App $appId should not change it's status to $status")
-        }
+        // A dead application should not be able to update status
+        if (!appRuntimeInfo.status.isInstanceOf[ApplicationTerminalStatus]) {
```

Comment:
Do we want to update status if not changed ? Can there be other "false alarm"s ?